File: stitch_nafaj_driver_dashboard/analyze.dart
import 'dart:io';
import 'dart:convert';

void main() {
  final dir = Directory('.');
  final List<Map<String, dynamic>> screens = [];

  for (var entity in dir.listSync()) {
    if (entity is Directory) {
      final name = entity.path.split(Platform.pathSeparator).last;
      if (name == 'nafaj' || name == '.dart_tool' || name == '.idea' || name.startsWith('.')) continue;
      
      final file = File('${entity.path}/code.html');
      
      if (file.existsSync()) {
        final content = file.readAsStringSync();
        
        var title = name;
        final lines = content.split('\n');
        for (var line in lines) {
          if (line.contains('<title>')) {
            final start = line.indexOf('<title>') + 7;
            final end = line.indexOf('</title>');
            if (end > start) {
              title = line.substring(start, end).trim();
            }
          }
        }
        
        final links = <String>{};
        for (var l in lines) {
            final parts = l.split('"');
            for (var p in parts) {
                if (p.endsWith('.html')) {
                    links.add(p.split('/').last);
                }
            }
            final parts2 = l.split("'");
            for (var p in parts2) {
                if (p.endsWith('.html')) {
                    links.add(p.split('/').last);
                }
            }
        }
        
        final parts = name.split('_');
        final className = parts.map((e) => e.isNotEmpty ? '${e[0].toUpperCase()}${e.substring(1)}' : '').join('') + 'Screen';
        
        screens.add({
          'dir_name': name,
          'class_name': className,
          'title': title,
          'links': links.toList(),
        });
      }
    }
  }

  File('analysis.json').writeAsStringSync(const JsonEncoder.withIndent('  ').convert(screens));
  print('Analysis complete. Found ${screens.length} screens.');
}

File: stitch_nafaj_driver_dashboard/analyze.py
import os
import json
import re

def analyze_app(base_dir):
    screens = []
    
    for item in os.listdir(base_dir):
        item_path = os.path.join(base_dir, item)
        if os.path.isdir(item_path):
            html_file = os.path.join(item_path, "code.html")
            if os.path.exists(html_file):
                with open(html_file, "r", encoding="utf-8") as f:
                    content = f.read()
                    
                # Extract title
                title_match = re.search(r'<title>(.*?)</title>', content)
                title = title_match.group(1) if title_match else item
                
                # Extract links
                links = re.findall(r'href=["\']([^"\']+\.html)["\']', content)
                links.extend(re.findall(r'onclick=["\']window\.location\.href=[\'"]([^\'"]+\.html)[\'"]', content))
                
                class_name = "".join(x.capitalize() for x in item.split("_")) + "Screen"
                
                screens.append({
                    "dir_name": item,
                    "class_name": class_name,
                    "title": title,
                    "links": list(set(links))
                })
    
    with open("analysis.json", "w") as f:
        json.dump(screens, f, indent=2)

if __name__ == "__main__":
    analyze_app(".")
